src/PIL/Jpeg2KImagePlugin.py

```diff
@@ -122,7 +122,8 @@ def _parse_codestream(fp):
     elif csiz == 4:
         mode = "RGBA"
     else:
-        mode = None
+        msg = "unable to determine JP2 image mode"
```

Comment:
```suggestion
        msg = "unable to determine J2K image mode"
```
https://github.com/python-pillow/Pillow/blob/99dd55324dcddc378341ca1dae75649c308bae43/src/PIL/Jpeg2KImagePlugin.py#L229-L230
might raise
https://github.com/python-pillow/Pillow/blob/99dd55324dcddc378341ca1dae75649c308bae43/src/PIL/Jpeg2KImagePlugin.py#L206-L207
So I think `_parse_codestream()` should use its codec name in an error.
https://github.com/python-pillow/Pillow/blob/99dd55324dcddc378341ca1dae75649c308bae43/src/PIL/Jpeg2KImagePlugin.py#L223-L224